pom.xml

```diff
@@ -231,6 +231,30 @@
                     </archive>
                 </configuration>
             </plugin>
+
+            <plugin>
+                <artifactId>maven-assembly-plugin</artifactId>
```

Comment:
It's been a while since I've done this, but the last time I needed a fatjar I used maven-shade-plugin instead.

The problem with maven-assembly-plugin is that it bundles *everything*, including test deps, and leaves you with little control of what goes in the jar and what doesn't.

The shade-plugin config is something like this:

```xml
<plugin>
    <groupId>org.apache.maven.plugins</groupId>
    <artifactId>maven-shade-plugin</artifactId>
    <version>3.0.0</version>
    <executions>
        <execution>
            <phase>package</phase>
            <goals>
                <goal>shade</goal>
            </goals>
            <configuration>
                <filters>
                    <filter>
                        <artifact>*:*</artifact>
                        <excludes>
                            <exclude>**/*.proto</exclude>
                            <exclude>META-INF/*.DSA</exclude>
                            <exclude>META-INF/*.RSA</exclude>
                            <exclude>META-INF/*.SF</exclude>
                        </excludes>
                    </filter>
                </filters>
                <finalName>ctlog-${project.version}-with-deps</finalName>
                <minimizeJar>true</minimizeJar>
            </configuration>
        </execution>
    </executions>
</plugin>
```

That leaves out test deps (and a few other undesirables as well).

Let me know your thoughts.